### PR TITLE
Spells/Priest: Protective Light Aura

### PR DESCRIPTION
**Pull request by cristianvnt**
**Issues addressed:**

[Protective Light](https://www.wowhead.com/spell=193063/protective-light) aura doesnt proc at all. It should proc from Flash Heal only.

**Tests performed:**
tested ingame
1. make a Priest.
2. have Protective Light selected in the talent tree.
3. use Flash Heal.

### CODE REVIEW

File: sql/updates/world/master/2024_01_28_03_world_priest_protective_light.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DELETE FROM `spell_proc` WHERE `SpellId` IN (193063);
+INSERT INTO `spell_proc` (`SpellId`,`SchoolMask`,`SpellFamilyName`,`SpellFamilyMask0`,`SpellFamilyMask1`,`SpellFamilyMask2`,`SpellFamilyMask3`,`ProcFlags`,`ProcFlags2`,`SpellTypeMask`,`SpellPhaseMask`,`HitMask`,`AttributesMask`,`DisableEffectsMask`,`ProcsPerMinute`,`Chance`,`Cooldown`,`Charges`) VALUES
+(193063,0x02,6,0x00000800,0x00000000,0x00000000,0x00000000,0x0,0x0,0x2,0x2,0x3,0x0,0x0,0,0,0,0); -- Protective Light
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (193063);
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES (193063, 'spell_pri_protective_light');
```
Comment (mdX7) [inline sql/updates/world/master/2024_01_28_03_world_priest_protective_light.sql:6]: we put the values on the next line, makes it easier to read
```suggestion
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(193063, 'spell_pri_protective_light');
```

File: src/server/scripts/Spells/spell_priest.cpp [hunk 2]
```diff
@@ -2788,8 +2789,31 @@ class spell_pri_vampiric_touch : public AuraScript
...
+    PrepareAuraScript(spell_pri_protective_light);
+
+    bool CheckEffectProc(const AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        return eventInfo.GetProcTarget() == GetCaster();
+    }
+
+    void HandleEffectProc(AuraEffect* auraEff, ProcEventInfo& eventInfo)
+    {
+        eventInfo.GetProcTarget()->CastSpell(GetCaster(), SPELL_PRIEST_PROTECTIVE_LIGHT_AURA, auraEff);
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_priest.cpp:2804]: would use `GetCaster()` here as its shorter and the same unit, thats just personal preference
```suggestion
        GetCaster()->CastSpell(GetCaster(), SPELL_PRIEST_PROTECTIVE_LIGHT_AURA, auraEff);
```

**Review by mdX7 — COMMENTED** [review]
nice job! :)

**Comment by Mematoru23** [comment]
alright
thanks for the code review

File: src/server/scripts/Spells/spell_priest.cpp [hunk 3]
```diff
@@ -2788,8 +2789,31 @@ class spell_pri_vampiric_touch : public AuraScript
     }
 };
 
+// 193063 - Protective Light (Aura)
+class spell_pri_protective_light : public AuraScript
+{
+    PrepareAuraScript(spell_pri_protective_light);
+
+    bool CheckEffectProc(const AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_priest.cpp:2797]: unused
```suggestion
    bool CheckEffectProc(const AuraEffect* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
```

File: src/server/scripts/Spells/spell_priest.cpp [hunk 4]
```diff
@@ -2788,8 +2789,31 @@ class spell_pri_vampiric_touch : public AuraScript
...
+    {
+        return eventInfo.GetProcTarget() == GetCaster();
+    }
+
+    void HandleEffectProc(AuraEffect* auraEff, ProcEventInfo& eventInfo)
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_priest.cpp:2802]: unused
```suggestion
    void HandleEffectProc(AuraEffect* auraEff, ProcEventInfo& /*eventInfo*/)
```

File: src/server/scripts/Spells/spell_priest.cpp (hunk 3 above)
Comment (Shauren) [inline src/server/scripts/Spells/spell_priest.cpp:2797]:
```suggestion
    bool CheckEffectProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
```

File: src/server/scripts/Spells/spell_priest.cpp [hunk 5]
```diff
@@ -2788,8 +2789,31 @@ class spell_pri_vampiric_touch : public AuraScript
     }
 };
 
+// 193063 - Protective Light (Aura)
+class spell_pri_protective_light : public AuraScript
+{
+    PrepareAuraScript(spell_pri_protective_light);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_priest.cpp:2795]: PrepareAuraScript was deprecated some time ago, it is not needed anymore

File: src/server/scripts/Spells/spell_priest.cpp (hunk 3 above)
Comment (Mematoru23) [reply src/server/scripts/Spells/spell_priest.cpp:2797]: oh right

File: src/server/scripts/Spells/spell_priest.cpp (hunk 5 above)
Comment (Mematoru23) [reply src/server/scripts/Spells/spell_priest.cpp:2795]: aight, didnt knew

**Pull request merged**